com.unity.ml-agents/Runtime/Analytics/InferenceAnalytics.cs

```diff
@@ -0,0 +1,275 @@
+using System;
+using System.Collections.Generic;
+using Unity.Barracuda;
+using Unity.MLAgents.Actuators;
+using Unity.MLAgents.Inference;
+using Unity.MLAgents.Policies;
+using Unity.MLAgents.Sensors;
+using UnityEngine;
+using UnityEngine.Analytics;
+
+#if UNITY_EDITOR
+using UnityEditor;
+using UnityEditor.Analytics;
+#endif
+
+
+namespace Unity.MLAgents.Analytics
+{
+    internal class InferenceAnalytics
+    {
+        const string k_VendorKey = "unity.ml-agents";
+        const string k_EventName = "ml_agents_inferencemodelset";
+
+        /// <summary>
+        /// Whether or not we've registered this particular event yet
+        /// </summary>
+        static bool s_EventRegistered = false;
+
+        /// <summary>
+        /// Hourly limit for this event name
+        /// </summary>
+        const int k_MaxEventsPerHour = 1000;
+
+        /// <summary>
+        /// Maximum number of items in this event.
+        /// </summary>
+        const int k_MaxNumberOfElements = 1000;
+
+        /// <summary>
+        /// Models that we've already sent events for.
+        /// </summary>
+        private static HashSet<NNModel> s_SentModels;
+
+        static bool EnableAnalytics()
+        {
+            if (s_EventRegistered)
+            {
+                return true;
+            }
+
+#if UNITY_EDITOR
```

Comment:
@surfnerd Following up from earlier review - this keeps the shape of the code the same for Editor and player. I think eventually we will want to send the same (or similar) event at runtime.

Reply:
I can move all the `#if UNITY_EDITOR` blocked code into smaller methods (like `IsAnalyticsEnabled()` below) if anyone feels strongly about it.

Reply:
LGTM